ui: display `musicResponsiveListItemFlexColumnRenderer` items in Home…

FineFindus opened the pull request:
…Screen

YouTube Music uses a song grid for some suggestions (especially when logged in) such as 'Trending in shorts', 'Covers', etc. Previously these were ignored, as the HomePage was only able to parse list sections. This expands the support for parsing and displaying the sections on the home screen.

<!-- Hey there. Thank you so much for improving OuterTune, and filling out the details. Having roughly the same layout helps everyone considerably :)-->

### What is it?

- [ ] New feature (user facing)
- [x] Update to existing feature (user facing)
- [ ] Bugfix (user facing)
- [ ] Codebase improvements or refactors (dev facing)
- [ ] Other

### Description of the changes in your PR

<!-- While bullet points are the norm in this section, feel free to write free-form text instead of a list -->


YouTube Music uses a song grid for some suggestions (especially when logged in) such as 'Trending in shorts', 'Covers', etc. Previously these were ignored, as the HomePage was only able to parse list sections. This expands the support for parsing and displaying the sections on the home screen.

### Before/After Screenshots/Screen Record

![Screenshot of the Covers and Remixes section](https://github.com/user-attachments/assets/e4a61fbb-8456-4898-b18c-09fb321c388b)

## Due diligence

<!-- Please mark WIP pull requests and "Draft" and only "Ready for review" once it is ready to be merged  -->

- [x] I read the [contribution guidelines](https://github.com/DD3Boh/OuterTune/blob/dev/CONTRIBUTING.md).

### Merge conflict resolution
Select only ONE. If you select none, or both, the first selection will used as your final choice

- [x] In the event of merge conflicts, I give permission for the developers to rebase or squash my code or apply merge
  conflict amendments as needed
- [ ] In the event of merge conflicts, I ***DO NOT*** give permission for the developers to modify my code to solve merge
  conflicts. I understand in the event of a merge conflict, I will be responsible to resolve merge conflicts in a way
  that adheres to the [contribution guidelines](https://github.com/DD3Boh/OuterTune/blob/dev/CONTRIBUTING.md)

<!-- This pull request template is based on Newpipe's:  https://github.com/TeamNewPipe/NewPipe/ -->

FineFindus committed:
ui: display `musicResponsiveListItemFlexColumnRenderer` items in HomeScreen

YouTube Music uses a song grid for some suggestions (especially when
logged in) such as 'Trending in shorts', 'Covers', etc.
Previously these were ignored, as the HomePage was only able to parse
list sections. This expands the support for parsing and displaying the
sections on the home screen.

DD3Boh commented:
This can give duplicated quick picks like shown here.

I'm not sure what the best solution would be, probably to just ignore the remote ones and use the local ones I would say.

![Screenshot_20250315-110542](https://github.com/user-attachments/assets/77eef3d3-82e7-4879-84f3-2feaaa3069f0)

FineFindus commented:
I would prefer to keep the remote ones (if available) as I've found them to be more accurate to my music taste, but I'm not strongly opposed to keeping the local ones. I think the bigger issue is how to filter them, as there doesn't seem to be any indicator for quick picks, besides the title, which is affected by translations.

DD3Boh commented:
I guess the easiest thing would be to hide the app's quick picks when you're logged in and just use YouTube ones, that way we could avoid any type of issues in theory. 

The only "issue" could be that quick picks won't be necessarily on top like they were now, but I don't necessarily see that as a problem.

FineFindus committed:
ui: load local items only if loginForBrowse is disabled

Some items can appear multiple times from both YouTube Music and the
local database, if loginForBrowse is enabled. This hides the local items
when the setting is enabled.

FineFindus commented:
Changed in https://github.com/OuterTune/OuterTune/pull/437/commits/ad657b3fe2a139db9dfca66dfdc40c81ff7a5b4e, which hides all local items, as I've noticed that the other items (e.g. forgotten favorites) can also appear duplicated, but I can revert that back to just quick picks if you prefer.

DD3Boh commented:
I'm not super sure about the forgotten favorites, because our local ones seem better than YouTube's.
When saying better I mean that they're more in the Quick picks style, while YouTube's forgotten favorites are just a list of some videos.

I will take a look at it later anyway, since as you mentioned there's probably no good way to filter that out specifically.
